packages/commonwealth/client/scripts/hooks/useDeferredConditionTriggerCallback.ts

```diff
@@ -0,0 +1,59 @@
+import { useCallback, useEffect, useRef } from 'react';
+
+type UseDeferredConditionTriggerCallbackProps = {
+  shouldRunTrigger?: boolean;
+};
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+type AllowAny = any; // we could get any type of args for a callback and its args/params so need to allow any
+
+// Usecase: When a UI element triggers a callback with args, but we want to run it only when an async condition
+// is satisfied. Example: If a button's callback should only run when user is logged in, but at current state
+// user is logged out, we can register the button callback with any args it passes and set `shouldRunTrigger` to
+// `true` or call the `trigger` method whenever the user login state becomes true to run the original callback.
+const useDeferredConditionTriggerCallback = ({
```

Comment:
@masvelio @Israellund - i couldn't think of a better name for this hook per the usecase, lmk if you guys have any ideas

Reply:
Lol, it's a mouthful but I think it works fine.